src/common/common/UrlUnicityChecker.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+
+import hashlib
+import os
+import pickle
+
+
+class UrlUnicityChecker(object):
+
+    def __init__(self, persist_file):
+        self.persist_file = persist_file
+        self.hashed_urls_set = set()
+        self._load()
```

Comment:
persist_file and hashed_urls fields (the fact that you use a set to check unicity) are implementation details. Thos fields should be private i think.